grazer/src/main/java/edu/asu/diging/grazer/web/FileUploadController.java

```diff
@@ -0,0 +1,86 @@
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import edu.asu.diging.grazer.core.domain.impl.FileImpl;
+import edu.asu.diging.grazer.core.fileupload.db.impl.FileUploadDatabaseConnection;
+
+@Controller
+public class FileUploadController {
+    
+    private final String FILE_EXTENSION = ".graphml";
+    
+    @Autowired
+    private FileUploadDatabaseConnection connection;
+    
+    @RequestMapping("/save-transformation")
+    public String uploadResources(@ModelAttribute FileImpl transformation,
+                                 Model model, @RequestParam CommonsMultipartFile[] files) {
+
+        List<byte[]> data = new ArrayList<byte[]>();
+        List<String> fileNames = new ArrayList<String>();
+        for (CommonsMultipartFile multipartFile : files) {
+            data.add(multipartFile.getBytes());
+            fileNames.add(multipartFile.getOriginalFilename()); 
+        }
+        
+        transformation.setfileNames(fileNames);
+        transformation.setData(data);
+        transformation.setDate(new Date());
+        transformation.setUploader(SecurityContextHolder.getContext().getAuthentication().getName());   
+        connection.save(transformation);
+        
+        model.addAttribute("transformation", transformation);
+        return "redirect:/transformation-input-form";    
+    }
+     
+    @RequestMapping(value = "transformation-input-form")
+    public String inputFile(Model model) {
+        model.addAttribute("transformation", new FileImpl());
+        model.addAttribute("fileList", connection.list());
+        return "fileUploadForm";
+    }
+    
+    @RequestMapping(value = "/download/{id}/{fileName}")
+    public String download(@PathVariable("id") int id, @PathVariable("fileName") String fileName, HttpServletResponse response) {
+        
+        fileName = fileName + FILE_EXTENSION;
+        FileImpl file = connection.get(id);
+        List<String> fileNames = file.getfileNames();
+        for(int i = 0; i < fileNames.size(); i++) {
+            if(fileName.equals(fileNames.get(i))) {
+                try {
+                    response.setContentType("APPLICATION/OCTET-STREAM");
+                    response.setHeader("Content-Disposition", "attachment;filename=\"" +fileName+ "\"");
+                    OutputStream out = response.getOutputStream();
+                    ByteArrayInputStream in = new ByteArrayInputStream(file.getData().get(i));
+                    IOUtils.copy(in, out);
+                    out.flush();
+                    out.close();
```

Comment:
all the logic to get the corresponding bytes for the filename should be done in a service class that sits between the controller and the database connection. there will probably be other cases where this logic is needed.